core/chains/evm/gas/models.go

```diff
@@ -261,32 +264,53 @@ func (e *evmFeeEstimator) L1Oracle() rollups.L1Oracle {
 	return e.EvmEstimator.L1Oracle()
 }
 
-func (e *evmFeeEstimator) GetFee(ctx context.Context, calldata []byte, feeLimit uint64, maxFeePrice *assets.Wei, opts ...feetypes.Opt) (fee EvmFee, chainSpecificFeeLimit uint64, err error) {
+func (e *evmFeeEstimator) GetFee(ctx context.Context, calldata []byte, feeLimit uint64, maxFeePrice *assets.Wei, toAddress *common.Address, opts ...feetypes.Opt) (fee EvmFee, chainSpecificFeeLimit uint64, err error) {
```

Comment:
After your changes, the return value from this function, param "chainSpecificFeeLimit", doesn't represent the absolute gas limit that caller provided. Instead, it represents what we estimate this specific attempt would consume, provided these fee params are used.

So better to rename the param to suggestedGasLimit. And in comment for this function, explain what this is.

Also, could you check the callers of this functions, whether they use this return value correctly, and next time when they gas bump, they pass in the input param "feeLimit" as original limit sent by caller, not the feeLimit used in previous attempt before a gas bump.

Reply:
Wonder if this is needed, since it's estimated only if `EstimateGasLimit` = false, otherwise `chainSpecificFeeLimit = feeLimit`